src/client/QXmppTransferManager.cpp

```diff
@@ -426,7 +426,7 @@ void QXmppTransferJob::terminate(QXmppTransferJob::Error cause)
     d->state = FinishedState;
 
     // close IO device
-    if (d->iodevice)
+    if (d->iodevice && d->iodevice->parent() == this)
```

Comment:
I don't like the idea of "having a parent equal to `this`" being a sign of "we have opened this device". Perhaps the `d`-implementation could be modified to have this flag explicitly?